### PR TITLE
Wrap CSL  to eliminate need for explicit pointer deallocation

### LINKED ISSUE #102: Figure out a way to `free` CSL objects
Unfortunately allocating memory with CSL requires us to `free` it afterwards. We might need to write some sort of pseudo garbage collector to register objects upon allocation and then free them in a monadic eliminator (e.g. `runQueryM`)

This is lower-priority and we have some time to think of the best way to approach this

### PR DESCRIPTION
**Pull request by amirmrad**
Closes #102  .

### Pre-review checklist

- [ ] All code has been formatted using our config (`make format`)
- [ ] Any new API features or modification of existing behavior is covered as defined in our [test plan](https://github.com/Plutonomicon/cardano-transaction-lib/blob/develop/doc/test-plan.md)
- [ ] The changelog has been updated under the `## Unreleased` header, using the appropriate sub-headings (`### Added`, `### Removed`, `### Fixed`), and the links to the appropriate issues/PRs have been included

### CODE REVIEW

File: test/Wallet/Cip30/SignData.js [hunk 1]
```diff
@@ -8,6 +8,7 @@ if (typeof BROWSER_RUNTIME != "undefined" && BROWSER_RUNTIME) {
   lib = require("@emurgo/cardano-message-signing-nodejs");
   csl = require("@emurgo/cardano-serialization-lib-nodejs");
 }
+lib = require("csl-runtime-gc")(lib);
```
Comment (jy14898) [inline test/Wallet/Cip30/SignData.js:11]: Needs to be done on both lib and csl

File: src/Internal/ApplyArgs.js [hunk 2]
```diff
@@ -9,6 +9,7 @@ if (typeof BROWSER_RUNTIME != "undefined" && BROWSER_RUNTIME) {
   lib = require("@emurgo/cardano-serialization-lib-nodejs");
   apply_args = require("apply-args-nodejs");
```
Comment (jy14898) [inline src/Internal/ApplyArgs.js:10]: Also to apply args?

File: src/Internal/ApplyArgs.js [hunk 3]
```diff
@@ -9,6 +9,7 @@ if (typeof BROWSER_RUNTIME != "undefined" && BROWSER_RUNTIME) {
   lib = require("@emurgo/cardano-serialization-lib-nodejs");
```
Comment (jy14898) [inline src/Internal/ApplyArgs.js:9]: What if we made a js file `src/Internal/CSL-wrapped.js` which imports and wraps the needed CSL, and then we import that from all foreign code? Then it's trivial to check all foreign imports use the wrapped one via a grep which searches for the unwrapped one

**Comment by klntsky** [comment]
Closing in favor of https://github.com/Plutonomicon/cardano-transaction-lib/pull/1386

File: src/Internal/ApplyArgs.js (hunk 3 above)
Comment (amirmrad) [reply src/Internal/ApplyArgs.js:9]: But won't spago move all the js files around during build? Using relative path requires doesn't seem straightforward.

**Comment by amirmrad** [comment]
Reopening since wasm bindgen approach failed with errors when testing. Falling back to wrapper solution.

**Review by klntsky — APPROVED** [review]

**Pull request merged**